File: vsa/examples/05-todo-list-py/src/contexts/tasks/delete-task/TaskDeletedEvent.py
"""Event representing delete-task completion"""

from typing import Optional
from event_sourcing import DomainEvent

class TaskDeletedEvent(DomainEvent):
    """Event representing task deletion
    
    Attributes:
        event_type: The event type identifier
        id: The task ID
        deleted_at: When the task was deleted
    """

    event_type: str = "TaskDeleted"
    id: str
    deleted_at: Optional[str] = None

File: vsa/examples/01-todo-list-ts/src/contexts/tasks/delete-task/TaskDeletedEvent.ts
import { BaseDomainEvent, EventType } from '@event-sourcing-platform/typescript';

/**
 * Event emitted when a task is deleted
 */
export class TaskDeletedEvent extends BaseDomainEvent {
  readonly eventType: EventType = 'TaskDeleted';
  readonly schemaVersion: number = 1;

  constructor(
    public readonly id: string,
    public readonly deletedAt: Date = new Date()
  ) {
    super();
  }
}

